test/copy.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+ * uriparser - RFC 3986 URI parsing library
+ *
+ * Copyright (C) 2007, Weijia Song <songweijia@gmail.com>
+ * Copyright (C) 2007, Sebastian Pipping <sebastian@pipping.org>
+ * Copyright (C) 2025, Máté Kocsis <kocsismate@php.net>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#include <uriparser/Uri.h>
+#include <uriparser/UriIp4.h>
+#include <gtest/gtest.h>
+#include <memory>
+#include <cstdio>
+#include <cstdlib>
+#include <cwchar>
+#include <string>
+
+using namespace std;
+
+namespace {
+	void testCopyUri(const char *const sourceUriString) {
+		UriParserStateA state;
+		UriUriA sourceUri;
+		state.uri = &sourceUri;
+
+		int result = uriParseSingleUriExA(&state, sourceUriString, sourceUriString + strlen(sourceUriString), NULL);
+		ASSERT_EQ(URI_SUCCESS, result);
+
+		UriUriA destUri;
+		ASSERT_EQ(URI_SUCCESS, uriCopyUriA(&destUri, &sourceUri));
+		ASSERT_EQ(URI_TRUE, uriEqualsUriA(&destUri, &sourceUri));
+
+		uriFreeUriMembersA(&sourceUri);
```

Comment:
```suggestion
		uriFreeUriMembersA(&sourceUri);
		uriFreeUriMembersA(&destUri);
```
Or am I missing something?

Reply:
Freeing happens on the call site, because the callers are asserting the components.

Reply:
@kocsismate the test is the call site — no? Please help me understand.